GetStream/stream-chat-android
# [3879] refactor SyncManager & EventHandlers

### 🎯 Goal

- Make `SyncManager` & `EventHandler` independent. 
- Remove `EventHandlerProvider` and let Plugin resolve dependencies
- Fix issues in `SyncManager`:
   - start syncing as soon as possible 
   - use `cids` from `SyncState` to restore
   - do not call `updateActiveChannels` if `updateActiveQueryChannels` fails


### 🧪 Testing

- Run any sample app
- Check the following scenarios:
   - resending the failed messages
   - backgrounding / foregrounding
   - enabling / disabling airplane mode


### ☑️Contributor Checklist

#### General

#### Code & documentation

### ☑️Reviewer Checklist

### 🎉 GIF

_Please provide a suitable gif that describes your work on this pull request_
